salt/states/boto_sns.py

```diff
@@ -228,23 +229,45 @@ def absent(
     profile
         A dict with region, key and keyid, or a pillar key (string)
         that contains a dict with region, key and keyid.
+
+    unsubscribe
+        If True, unsubscribe all subcriptions to the SNS topic before
+        deleting the SNS topic
     '''
     ret = {'name': name, 'result': True, 'comment': '', 'changes': {}}
 
     is_present = __salt__['boto_sns.exists'](name, region, key, keyid, profile)
 
     if is_present:
-        if __opts__['test']:
-            ret['comment'] = 'AWS SNS topic {0} is set to be removed.'.format(
-                name)
+        subscriptions = __salt__['boto_sns.get_all_subscriptions_by_topic'](
+                            name, region, key, keyid, profile
+                        ) if unsubscribe else []
+        failed_unsubscribe_subscriptions = []
+
+        if __opts__.get('test'):
+            ret['comment'] = (
+                'AWS SNS topic {0} is set to be removed.  '
+                '{1} subscription(s) will be removed.'.format(name, len(subscriptions))
+            )
             ret['result'] = None
             return ret
+
+        for subscription in subscriptions:
+            unsubscribed = __salt__['boto_sns.unsubscribe'](name, subscription['SubscriptionArn'])
```

Comment:
it is preserved since subscriptions list will be initialized to [] if unsubscribe is False up near line 243-245..